CI: Fedora/CentOS based jobs are failing after Fedora 35 release

There are jobs which use the image-builder (`kata-containers/tools/osbuilder/image-builder`) to build the guest image. The image-builder is actually a container built out of Fedora. Currently it uses `fedora:latest`; Fedora 35 was released yesterday and that started to fail the build of the image-builder image (the fedora 35 dnf mirrors are broken).

Here is an error log:
```
00:58:07 + podman build --build-arg http_proxy= --build-arg https_proxy= -t image-builder-osbuilder /tmp/jenkins/workspace/kata-containers-2.0-centos-8-CRIO_K8S-main-baseline/go/src/github.com/kata-containers/kata-containers/tools/osbuilder/image-builder
00:58:07 STEP 1: FROM registry.fedoraproject.org/fedora:latest
00:58:07 Trying to pull registry.fedoraproject.org/fedora:latest...
00:58:08 Getting image source signatures
00:58:08 Copying blob sha256:60d974a73230da76eb030c3439d8c859ada8f47a2a31e2a99516f24378edcb37
00:58:10 Copying config sha256:d08dfa41ca0b4449b2da67a85fee01d6d5901f1cb700edf6767a13d2de7dc252
00:58:10 Writing manifest to image destination
00:58:10 Storing signatures
00:58:12 STEP 2: RUN [ -n "$http_proxy" ] && sed -i '$ a proxy='$http_proxy /etc/dnf/dnf.conf ; true
00:58:13 --> 9455ff83fc7
00:58:13 STEP 3: RUN dnf install -y qemu-img parted gdisk e2fsprogs gcc xfsprogs findutils
00:58:13 Fedora 35 - x86_64                              0.0  B/s |   0  B     00:00    
00:58:13 Errors during downloading metadata for repository 'fedora':
00:58:13   - Curl error (6): Couldn't resolve host name for https://mirrors.fedoraproject.org/metalink?repo=fedora-35&arch=x86_64 [getaddrinfo() thread failed to start]
00:58:13 Error: Failed to download metadata for repo 'fedora': Cannot prepare internal mirrorlist: Curl error (6): Couldn't resolve host name for https://mirrors.fedoraproject.org/metalink?repo=fedora-35&arch=x86_64 [getaddrinfo() thread failed to start]
00:58:14 Error: error building at STEP "RUN dnf install -y qemu-img parted gdisk e2fsprogs gcc xfsprogs findutils": error while running runtime: exit status 1
00:58:14 make: *** [Makefile:130: /tmp/jenkins/workspace/kata-containers-2.0-centos-8-CRIO_K8S-main-baseline/go/src/github.com/kata-containers/kata-containers/tools/osbuilder/kata-containers.img] Error 125
00:58:14 [install_kata_image.sh:50] ERROR: sudo -E USE_DOCKER= DISTRO=ubuntu make -e image
00:58:14 [install_kata.sh:22] ERROR: .ci/install_kata_image.sh 
00:58:14 [jenkins_job_build.sh:172] ERROR: ci/setup.sh
```

--- a/tools/osbuilder/image-builder/Dockerfile
+++ b/tools/osbuilder/image-builder/Dockerfile
@@ -3,7 +3,7 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 ARG IMAGE_REGISTRY=registry.fedoraproject.org
-FROM ${IMAGE_REGISTRY}/fedora:latest
+FROM ${IMAGE_REGISTRY}/fedora:34
 
 RUN [ -n "$http_proxy" ] && sed -i '$ a proxy='$http_proxy /etc/dnf/dnf.conf ; true
 

